numpy/lib/tests/test_format.py

```diff
@@ -536,17 +536,14 @@ def test_python2_python3_interoperability():
         data = np.load(path)
     assert_array_equal(data, np.ones(2))
 
-def test_pickle_python2_python3():
```

Comment:
What's happening with these code removals?

Reply:
I had to update array files in the `numpy/lib/tests/data/` folder, because otherwise these tests are failing with "`numpy.core.multiarray` doesn't exist".  
When you `cat` these `.npy` and `.npz` files, in their second lines you will see `numpy._core.multiarray`. Numpy's module path is there so I had to recreate these files with `np.save`.  

For `Python 2` files (`py2-objarr.npy` and `py2-objarr.npz`) I would need to be able to run the codebase with Python 2 to also get updated versions of these zips, but as I understand Python 2 is no longer supported. 

Therfore I decided to drop test cases where we test against Python 2. Is it correct? Or should we get them anyway?

P.S. I tried to manually change `core` to `_core` in vim for Python 2 `npy` and `npz` files but then, as expected, I get a "checksum error" on loading them.

Reply:
I already had to do such `.npy`/`.npz` file updates recently: https://github.com/PyWavelets/pywt/pull/679. It seems to have been a problem in numpy 1.25.2 already.

This change is larger, and hence more "saved in python 2.X" files stop working. They'll probably start working again if you put back `numpy/core/__init__.py` with a deprecation warning. But either way, eventually files with object arrays are going to stop working indeed. Technically pickled data isn't portable between Python versions at all IIRC. I don't think it affects too many users of `.npy` files, only those with object arrays.

@rkern do you have any objections to dropping these tests?

Reply:
Yes, pickled data is portable between Python versions, as long as both versions support the same protocol level. [It's one of the defining differences between it and `marshal`](https://docs.python.org/3/library/pickle.html#comparison-with-marshal).

I do object to dropping these tests. I would like these old files to continue to work, as they exist. If you want updated versions of some of these files to test with the `numpy._core` names, then please make them *in addition* to the existing files. How the unpickler resolves names can be customized by overriding [`Unpickler.find_class()`](https://docs.python.org/3/library/pickle.html#pickle.Unpickler.find_class), and we should do so where we do use unpickling in the NPY/NPZ formats in order to follow this renaming.

Whether we expose this customized `Unpickler` for others to use for their own unpickling to handle this refactoring, I am neutral on. But for the NPY/NPZ formats, we abstracted away many of these details from the user, and I think we should take this relatively small step to ensure that our internal refactorings do not invalidate their data.

Reply:
Ah, that's a good idea, thanks. +1 for an unpickler.

Reply:
Sure! I reintroduced python2 pickle tests and managed to get them passing by extending `core` shim that is used when `npy`/`npz` file was pickled with `numpy.core` module.